aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/ComparisonUtils.java

```diff
@@ -0,0 +1,35 @@
+package software.amazon.codeartifact.repository;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+public class ComparisonUtils {
+
+    public static boolean willAddUpstreams(final ResourceModel desiredModel, final ResourceModel prevModel) {
+        return prevModel.getUpstreams() == null && desiredModel.getUpstreams() != null;
+    }
+
+    public static boolean willNotUpdateUpstreams(final ResourceModel desiredModel, final ResourceModel prevModel) {
+        return upstreamsAreEqual(desiredModel, prevModel);
+    }
+
+    public static boolean willUpdateUpstreams(final ResourceModel desiredModel, final ResourceModel prevModel) {
+        return !upstreamsAreEqual(desiredModel, prevModel);
+    }
```

Comment:
purely a nit, but by adding both the true and false methods for the same logical thing, you might have to update both methods if you are changing what this condition means, instead of having a single place that defines the true case, and then when you use it you just `!` it.

Reply:
I see, ok will update